src/components/CodeShape/index.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+
+import styles from './styles.module.scss';
+
+export enum Token {
+  code = 'code',
+  whitespace = 'whitespace',
+}
+
+// This represents the shape of one or more tokens.
+export type TokenShape = {
+  count: number;
+  percentOfWidth: number;
+  token: Token.code | Token.whitespace;
+};
+
+// This is a collection of shapes for one line of code.
+export type LineShapes = {
+  line: number;
+  tokens: TokenShape[];
+};
+
+// These are shape collections for all lines of code.
+export type AllLineShapes = LineShapes[];
+
+export const generateLineShapes = (
```

Comment:
Could we move this function to `CodeShape/utils.tsx`? I don't see it used in the component and that's (in part) why I thought the component would render shapes for all the lines of a file.